pkg/controller/ses/template/setup.go

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright 2021 The Crossplane Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package template
+
+import (
+	"context"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"k8s.io/client-go/util/workqueue"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+
+	svcsdk "github.com/aws/aws-sdk-go/service/ses"
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	"github.com/crossplane/crossplane-runtime/pkg/event"
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+	"github.com/crossplane/crossplane-runtime/pkg/meta"
+	"github.com/crossplane/crossplane-runtime/pkg/ratelimiter"
+	"github.com/crossplane/crossplane-runtime/pkg/reconciler/managed"
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+
+	svcapitypes "github.com/crossplane/provider-aws/apis/ses/v1alpha1"
+	aws "github.com/crossplane/provider-aws/pkg/clients"
+)
+
+// SetupTemplate adds a controller that reconciles Template.
+func SetupTemplate(mgr ctrl.Manager, l logging.Logger, rl workqueue.RateLimiter, poll time.Duration) error {
+	name := managed.ControllerName(svcapitypes.TemplateGroupKind)
+	opts := []option{
+		func(e *external) {
+			e.postCreate = postCreate
+			e.preObserve = preObserve
+			e.postObserve = postObserve
+			e.preDelete = preDelete
+			e.isUpToDate = isUpToDate
+
+		},
+	}
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		WithOptions(controller.Options{
+			RateLimiter: ratelimiter.NewDefaultManagedRateLimiter(rl),
+		}).
+		For(&svcapitypes.Template{}).
+		Complete(managed.NewReconciler(mgr,
+			resource.ManagedKind(svcapitypes.TemplateGroupVersionKind),
+			managed.WithInitializers(managed.NewDefaultProviderConfig(mgr.GetClient())),
+			managed.WithExternalConnecter(&connector{kube: mgr.GetClient(), opts: opts}),
+			managed.WithPollInterval(poll),
+			managed.WithLogger(l.WithValues("controller", name)),
+			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
+}
+
+func preObserve(_ context.Context, cr *svcapitypes.Template, obj *svcsdk.GetTemplateInput) error {
+	obj.TemplateName = aws.String(meta.GetExternalName(cr))
+	cr.Spec.ForProvider.Template.TemplateName = obj.TemplateName
```

Comment:
I *think* you should add this field to ignore in code gen. Look at lambda for example, the field `FunctionName` is ignored, so it's only part of external-name and not from the cr.
https://github.com/crossplane/provider-aws/blob/36e425acdcb52a670f78ab5e37b8f81864d3f393/pkg/controller/lambda/function/setup.go#L87-L90

https://github.com/crossplane/provider-aws/blob/36e425acdcb52a670f78ab5e37b8f81864d3f393/apis/lambda/v1alpha1/generator-config.yaml#L1-L3

Reply:
The `CreateTemplateInput` struct contains a `*Template` which contains the `TemplateName`.

Do you suggest adding TemplateName to ignore field_paths like this?
```
ignore:
  field_paths:
    - CreateTemplateInput.Template.TemplateName
```


I tried adding the same. The resultant code generated after re running make generate does not have the Template Parameters in the ForProvider i.e the CRD does not accept the `Template_SDK struct` that contains `HTMLPart, SubjectPart, TemplateName, TextPart`

Reply:
I think there is a need to update the code-generator, it does not support three levels like `Foo.Bar.Baz`.

I made a (draft) PR here: https://github.com/aws-controllers-k8s/code-generator/pull/119

Reply:
@muvaf points out that this can be achieved already using

```yaml
ignore:
  field_paths:
    - Template.TemplateName
```